### PR TITLE
Model relationships setup

### PR DESCRIPTION
**Pull request by cemccabe**
## What does this merge do? Check all that apply.
- [x] Adds a new feature
- [ ] Improves an existing feature
- [ ] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)

## How has this been tested?
Model and requests tested in RSpec
## Description:
Created Users and Favorites tables, as well as created index, create, and destroy functionality for a Favorite request
## Issues:
No known issues at this time -- we will work with FE when they access the endpoints to determine any issues
## Where to find Changes:
Files: Controllers, models, specs
Methods:
Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have used tests to prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my change

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,11 +96,19 @@ Utilize an Agile project management process to hone skill and comfort with build
 * Ruby 2.7.4
 * Rails 5.2.8
```
Comment (Kerynn) [inline README.md:98]: This branch has a few changes to readme, but just a reminder we'll need to add the endpoints!

**Review by Kerynn — APPROVED** [review]
@Kledin85 - We couldn't think of a reason for needing to update a favorite, so only have create, index, and destroy in our favorites controller. Unless you can think of a reason for any other crud?

**Review by Kledin85 — APPROVED** [review]
The read me is looking great!

File: README.md (hunk 1 above)
Comment (cemccabe) [reply README.md:98]: For future reference, here are our current endpoints that need documenting:

GET `/api/v1/campsites/<CAMPSITE ID>`
GET `/api/v1/campsites?state_code=<STATE CODE>`
GET `/api/v1/campsites?q=<CAMPSITE NAME>`
GET `/api/v1/campsites?park_name=<PARK NAME>`

GET `/api/v1/favorites?user_id=1`
POST `/api/v1/favorites?user_id=1`
DELETE `/api/v1/favorites/<FAVORITE_ID>`

**Review by CassandraGoose — COMMENTED** [review]
So, I'm not a Rails person - but here's what I assume is happening: 

- when the FE makes a request to the /favorites endpoint, they will be able to create a favorite and delete a favorite and this will be based on what is passed as the id parameter (which finds a user matching that id). 

If I have that right, then I think that's a solid flow for your app! It will allow y'all to see favorites of various users without the need for auth/auth. 🎉 

One thing I'd like to note is that I noticed that favorites have the campsite name, id, details, and image in the records. I don't know if your plan was to store all campsites in general - but that could mean there's some redundant data in the db. For example, if you're already saving campsite information in campsite records then it's a little redundant to also save all that data into the favorites records. More likely out the wild, there would be some kind of join table or other system to help normalize that data. This isn't a requirement or request - just food for thought, especially since I don't know y'all's full plan and context. 

I'm not going to throw an approval or change request - I'll trust y'all to do that based on the other comments I see in this PR. On my end, I think this will work for you! Nice work.

**Pull request merged**